src/frontend/src/binder/expr/function.rs

```diff
@@ -354,6 +354,7 @@ impl Binder {
                     ]),
                 ),
                 ("date_trunc", raw_call(ExprType::DateTrunc)),
+                ("date_part", raw_call(ExprType::Extract)),
```

Comment:
The subtle difference between `date_part` and `extract` in PostgreSQL is the former returns `double precision` (approx) while the latter returns `decimal` (exact).